Explorer/Assets/DCL/Multiplayer/Connections/Rooms/Connective/ConnectiveRoom.cs

```diff
@@ -61,18 +62,23 @@ CycleStepDelegate runConnectCycleStepAsync
 
         public void Start()
         {
-            if (CurrentState() is not IConnectiveRoom.State.Sleep)
+            if (CurrentState() is not IConnectiveRoom.State.Stopped)
                 throw new InvalidOperationException("Room is already running");
 
             roomState.Set(IConnectiveRoom.State.Starting);
             RunAsync().Forget();
         }
 
-        public void Stop()
+        public async UniTask StopAsync()
         {
-            roomState.Set(IConnectiveRoom.State.Sleep);
-            cancellationTokenSource?.Cancel();
-            cancellationTokenSource?.Dispose();
+            if (CurrentState() is IConnectiveRoom.State.Stopped or IConnectiveRoom.State.Stopping)
+                throw new InvalidOperationException("Room is already stopped");
```

Comment:
I wonder what will happen in the upper flow if the exception is thrown?

Reply:
It's invalid request to double stop the room like a double disconnect of websocket